src/robot/utils/typehints.py

```diff
@@ -0,0 +1,31 @@
+#  Copyright 2008-2015 Nokia Networks
+#  Copyright 2016-     Robot Framework Foundation
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+
+from typing import Any, Callable, TypeVar, Generic
+
+T = TypeVar('T', bound=Callable[..., Any])
+
+
+class copy_signature(Generic[T]):
+    """A decorator that applies the signature of `F` to any function that it decorates
+    see https://github.com/python/typing/issues/270#issuecomment-555966301 for source
+    and discussion.
+    """
+
+    def __init__(self, target: T) -> None: ...
+
+    def __call__(self, wrapped: Callable[..., Any]) -> T:
+        return wrapped  # type: ignore
```

Comment:
Isn't this now unnecessarily complicated for our needs? Based on my test this seems to be enough:

```python
def copy_signature(target: T) -> Callable[..., T]:
    def decorator(func):
        return func
    return decorator
```

I left out typing from `decorator` on purpose. It doesn't seem to be needed, and it wouldn't anyway match the actual parameters or the return type.

Reply:
I don't mind either way. But I think the class-based approach would be slightly more performant since It's not creating any functions.

Reply:
In the class based approach an instance is created and I believe it's slower. I don't think performance matter either way so a simpler approach is probably better.